cpuminer.go

```diff
@@ -251,7 +251,7 @@ func (m *CPUMiner) solveBlock(msgBlock *wire.MsgBlock, ticker *time.Ticker, quit
 	// Note that the entire extra nonce range is iterated and the offset is
 	// added relying on the fact that overflow will wrap around 0 as
 	// provided by the Go spec.
-	for extraNonce := uint64(0); extraNonce < maxExtraNonce; extraNonce++ {
+	for extraNonce := uint64(0); extraNonce <= maxExtraNonce; extraNonce++ {
```

Comment:
So this is actually correct and the other case is incorrect.  Go automatically overflows back to zero, so this (and the other case) would create infinite loops when a solution isn't found.  It's never been hit in practice because this is only every used with difficulties the result in a solution within the regular nonce space.

This can be shown with the following code:
```Go
package main

import (
	"fmt"
)

func main() {
	const maxFoo = ^uint8(0) //  2^8 - 1
	var numWraps int
	for i := uint8(0); i <= maxFoo; i++ {
		if i == 0 {
			numWraps++
		}
		if numWraps == 2 {
			fmt.Println("Wrapped around")
			return
		}
	}
}
```

Reply:
Good catch, I should have realized this!  I'll be afk for a few days, but will post an update once I'm back.

Reply:
Opened a separate PR#1865 for the new fix.